File: shiro_test/util/src/main/java/com/oik/util/dto/UserDTO.java
package com.oik.util.dto;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
public class UserDTO {
    private Long userId;
    private String username;
    private String deptId;
    private String email;
    private String mobile;
    private String description;
    private String token;
    private String ssex;
    private String avatar;
    private String ip;
    private LocalDateTime lastLoginTime;

    public UserDTO(String username) {
        this.username = username;
    }
}

File: v3-ui-ts/src/store/UserDto.ts
import {defineStore} from 'pinia'

export const User = defineStore('userDto', {

    state: () => ({
        userDto: {
            userId: '',
            username: '',
            deptId: '',
            email: '',
            mobile: '',
            description: '',
            token: '',
            ssex: '',
            avatar: '',
            ip: '',
        }
    }),

    getters: {
        getUser: (userDto) => userDto
    },

    actions: {
        setUserDto(userDto) {
            this.userDto = userDto
        },
    },
    persist: {
        enabled: true, // 开启数据缓存
    }

})
